EdgeOfEmpireBot/EdgeOfEmpireBot/Services/DataService/DataService.cs

```diff
@@ -104,6 +105,16 @@ public async Task<T> ReadFromFile<T>(string fileName)
         return JsonConvert.DeserializeObject<T>(await File.ReadAllTextAsync(path))!;
     }
 
+    /// <inheritdoc/>
+    public async Task UpsertToJsonFile<TKey, TValue>(string fileName, (TKey Key, TValue Value) contentToWrite) where TKey : notnull
```

Comment:
I don't deal with databases enough to know upsert ahah

Reply:
Insert/Update

Reply:
I know I looked up ahaha